src/main/scala/org/broadinstitute/dsde/workbench/sam/service/UserService.scala

```diff
@@ -115,6 +115,28 @@ class UserService(val directoryDAO: DirectoryDAO, val cloudExtensions: CloudExte
       case None => IO(user)
     }
 
+  def updateUserCrud(userId: WorkbenchUserId, request: AdminUpdateUserRequest, samRequestContext: SamRequestContext): IO[Option[SamUser]] =
+    openTelemetry.time("api.v1.user.updateUserCrud.time", API_TIMING_DURATION_BUCKET) {
+      directoryDAO.loadUser(userId, samRequestContext).flatMap {
+        case Some(user) =>
+          // validate all fields to be updated
+          var errorReports = Seq[ErrorReport]()
+          request.email.foreach(email => errorReports = errorReports ++ validateEmail(email, blockedEmailDomains))
+          if (errorReports.nonEmpty) {
+            IO.raiseError(new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, "invalid user update", errorReports)))
+          } else { // apply all updates
+            var updatedUser = user
+            request.email.foreach { email =>
+              directoryDAO.updateUserEmail(userId, email, samRequestContext)
+              updatedUser = user.copy(email = email)
```

Comment:
This feels a little smelly, I think id prefer to do a select for the new user state from the db just in case there are two concurrent requests coming in. This endpoint is for admins only and shouldn't see a lot of traffic so i think two separate db calls and transactions is fine.

Reply:
Yeah I was sort of going back and forth about this. What do you think about using the `RETURNING` parameter in the SQL statement (https://www.postgresql.org/docs/9.5/sql-update.html). That way we can get the record directly from the db without creating a new call?

Reply:
I am noticing that we do use this same pattern (using the request object to construct the response directly) when we do a createUser, but I do think there is a better solution. I think it might be a good ticket to open if I can find a more general example of returning the status of DB calls. Currently, most of our DAO methods return a Unit which I do not really like the design of so I think this is worth looking into

Reply:
@sjkobori Okay yeah this is a can of worms agreed lets create a ticket.